gpu/gpu_info_cuda.c

```diff
@@ -82,121 +86,283 @@ void cuda_init(char *cuda_lib_path, cuda_init_resp_t *resp) {
   }
 }
 
-void cuda_check_vram(cuda_handle_t h, mem_info_t *resp) {
-  resp->err = NULL;
+/**
+ * Allocates and initializes a new device map for tracking NVML devices and
+ * their MIG instances. This map is structured to hold references to both
+ * primary CUDA devices and up to 8 MIG device handles per primary device,
```

Comment:
I think we should refactor this to handle dynamic MIG device counts.  Perhaps as simple as `calloc(<maxmigdevicecount>, sizeof(nvmlDevice_t))`

Reply:
I considered that, and I think it's faster and more readable to blindly allocate exactly 9*sizeof(nvmlDevice_t) knowing that it can't be a larger number than that, rather than iterating over all the devices to determine the number than going back to allocate, and then assign.  It will use extra memory (potentially 72 extra bytes per host GPU).

Reply:
> knowing that it can't be a larger number than that

Where did you see that?  From my read of the docs, the number looks to be dynamic, and as new GPUs come out, this may increase over time.  I found mention of A100's supporting 10 virtual devices.

Reply:
Also, at this layer of the stack, I'm less concerned about a little wasted space in bytes if it aids readability and maintainability of the code longer term.